src/NuGet.Server/Infrastructure/ServerPackageRepository.cs

```diff
@@ -539,6 +542,7 @@ public void ClearCache()
 
                 _serverPackageStore.Clear();
                 _serverPackageStore.Persist();
+                _needsRebuild = true;
```

Comment:
It uses `Monitor.Enter` on `_syncLock` which is the same thing that `lock` does.
https://github.com/NuGet/NuGet.Server/blob/813ea519ae5d1ad56af7b7ed7428a429357b14dd/src/NuGet.Server/Infrastructure/ServerPackageRepository.cs#L731